src/ts/HttpReader.ts

```diff
@@ -0,0 +1,220 @@
+import { flatbuffers } from 'flatbuffers';
+
+import { Rect, calcTreeSize, DEFAULT_NODE_SIZE, NODE_ITEM_LEN, streamSearch} from './packedrtree';
+import { magicbytes, SIZE_PREFIX_LEN } from './constants';
+import Logger from './Logger';
+import HeaderMeta from './HeaderMeta'; 
+import { Feature } from './feature_generated';
+
+export class HttpReader {
+    private headerClient: BufferedHttpRangeClient;
+    private _featureClient?: BufferedHttpRangeClient;
```

Comment:
One place we diverged from the rust client...

In rust we parse the entire index, then start parsing features.
whereas in JS we were yielding a feature as soon as we got one, while still working through the index.

The repercussion there is that we'll be jumping back and forth between reading the index and reading the features, so we need separate buffers.

Reply:
Not sure of the pros and cons here. I guess there could be cases where forward seek only could be desirable but I suspect that I/O pattern/requirement is disappearing fast. So I'm leaning towards accepting this.

Reply:
FWIW, I mostly went this way just because its' what the JS code was already doing.

Some pros that come to mind with this approach though:
1. time to first feature is minimized
2. we don't have to hold all the search results (`Array<[byteOffsets, featureIdx]>`) in memory at once

cons:
1. two buffers means more memory... but it's a fixed amount rather than point (2.) above.
2. seek times as you mentioned, though I imagine once network latency is involved, that disk seek times are less significant.

Reply:
Thanks for the explanation, I'm fine with keeping it as is.